src/evict/evict_lru.c

```diff
@@ -584,8 +584,10 @@ __evict_update_work(WT_SESSION_IMPL *session)
     if (!__evict_queue_empty(cache->evict_urgent_queue, false))
         LF_SET(WT_CACHE_EVICT_URGENT);
 
-    if (F_ISSET(conn, WT_CONN_HS_OPEN) && __wt_hs_get_btree(session, &hs_tree) == 0)
+    if (F_ISSET(conn, WT_CONN_HS_OPEN) && __wt_hs_get_btree(session, &hs_tree) == 0) {
         cache->bytes_hs = hs_tree->bytes_inmem;
+        cache->bytes_hs_dirty = hs_tree->bytes_dirty_intl + hs_tree->bytes_dirty_leaf;
```

Comment:
Could you add a comment, something like "TODO: We are caching the cache usage values associated with the history store because the history store dhandle isn't always available to eviction. Keeping potentially out-of-date values could lead to surprising bugs in the future."

And create a ticket to capture the work of fixing this via being able to access the history store btree.

Reply:
I have created WT-7096 to track future work for improving the cache stat collection from the history store.